metacat-main/src/main/java/com/netflix/metacat/main/api/RequestWrapper.java

```diff
@@ -216,10 +219,23 @@ public QualifiedName qualifyName(final Supplier<QualifiedName> nameSupplier) {
         } finally {
             final long duration = registry.clock().wallTime() - start;
             log.info("### Time taken to complete {} for {} is {} ms", resourceRequestName, name, duration);
+            tryAddTableTypeTag(tags, name);
             this.registry.timer(requestTimerId.withTags(tags)).record(duration, TimeUnit.MILLISECONDS);
         }
     }
 
+    private static void tryAddTableTypeTag(final Map<String, String> tags, final QualifiedName qualifiedName) {
+        final MetacatRequestContext context = MetacatContextManager.getContext();
+        if (context == null) {
+            return;
+        }
+        String tableType = context.getTableType(qualifiedName);
+        if (tableType == null) {
```

Comment:
probably make this null or empty. StringUtils.isBlank()

Reply:
fixed